tensorflow/contrib/eager/python/examples/generative_examples/dcgan.ipynb

```diff
@@ -1,324 +1,405 @@
-        "    self.batchnorm3 = tf.keras.layers.BatchNormalization()\n",
+        "def make_generator_model():\n",
+        "    model = tf.keras.Sequential()\n",
+        "    model.add(tf.keras.layers.Dense(7*7*256, use_bias=False, input_shape=(100,)))\n",
+        "    model.add(tf.keras.layers.BatchNormalization())\n",
+        "    model.add(tf.keras.layers.LeakyReLU())\n",
+        "      \n",
+        "    model.add(tf.keras.layers.Reshape((7, 7, 256)))\n",
+        "    assert model.output_shape == (None, 7, 7, 256)\n",
         "    \n",
-        "    self.conv3 = tf.keras.layers.Conv2DTranspose(1, (5, 5), strides=(2, 2), padding='same', use_bias=False)\n",
-        "\n",
-        "  def call(self, x, training=True):\n",
-        "    x = self.fc1(x)\n",
-        "    x = self.batchnorm1(x, training=training)\n",
-        "    x = tf.nn.relu(x)\n",
-        "\n",
-        "    x = tf.reshape(x, shape=(-1, 7, 7, 64))\n",
+        "    model.add(tf.keras.layers.Conv2DTranspose(128, (5, 5), strides=(1, 1), padding='same', use_bias=False))\n",
+        "    assert model.output_shape == (None, 7, 7, 128)  \n",
+        "    model.add(tf.keras.layers.BatchNormalization())\n",
+        "    model.add(tf.keras.layers.LeakyReLU())\n",
         "\n",
-        "    x = self.conv1(x)\n",
-        "    x = self.batchnorm2(x, training=training)\n",
-        "    x = tf.nn.relu(x)\n",
+        "    model.add(tf.keras.layers.Conv2DTranspose(64, (5, 5), strides=(2, 2), padding='same', use_bias=False))\n",
+        "    assert model.output_shape == (None, 14, 14, 64)    \n",
+        "    model.add(tf.keras.layers.BatchNormalization())\n",
+        "    model.add(tf.keras.layers.LeakyReLU())\n",
         "\n",
-        "    x = self.conv2(x)\n",
-        "    x = self.batchnorm3(x, training=training)\n",
-        "    x = tf.nn.relu(x)\n",
+        "    model.add(tf.keras.layers.Conv2DTranspose(1, (5, 5), strides=(2, 2), padding='same', use_bias=False, activation='tanh'))\n",
+        "    assert model.output_shape == (None, 28, 28, 1)\n",
+        "  \n",
+        "    return model"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "colab_type": "text",
+        "id": "D0IKnaCtg6WE"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "### The Discriminator model\n",
         "\n",
-        "    x = tf.nn.tanh(self.conv3(x))  \n",
-        "    return x"
+        "The discriminator is responsible for distinguishing fake images from real images. It's similar to a regular CNN-based image classifier."
       ]
     },
     {
+      "metadata": {
+        "id": "dw2tPLmk2pEP",
+        "colab_type": "code",
+        "colab": {}
+      },
       "cell_type": "code",
+      "source": [
+        "def make_discriminator_model():\n",
+        "    model = tf.keras.Sequential()\n",
```

Comment:
Optional: Since there are no asserts here, the original version (where the list was provided to the constructor) works just as well. Your call either way.

Reply:
I left it as is because it's easier for the reader to experiment with assert, print or model.summary() etc. if they want to.